torchvision/datasets/kinetics.py

```diff
@@ -89,3 +246,68 @@ def __getitem__(self, idx):
             video = self.transform(video)
 
         return video, audio, label
+
+
+class Kinetics400(Kinetics):
+    """
+    `Kinetics-400 <https://deepmind.com/research/open-source/open-source-datasets/kinetics/>`_
+    dataset.
+
+    Kinetics-400 is an action recognition video dataset.
+    This dataset consider every video as a collection of video clips of fixed size, specified
+    by ``frames_per_clip``, where the step in frames between each clip is given by
+    ``step_between_clips``.
+
+    To give an example, for 2 videos with 10 and 15 frames respectively, if ``frames_per_clip=5``
+    and ``step_between_clips=5``, the dataset size will be (2 + 3) = 5, where the first two
+    elements will come from video 1, and the next three elements from video 2.
+    Note that we drop clips which do not have exactly ``frames_per_clip`` elements, so not all
+    frames in a video might be present.
+
+    Internally, it uses a VideoClips object to handle clip creation.
+
+    Args:
+        root (string): Root directory of the Kinetics-400 Dataset. Should be structured as follows:
+
+            .. code::
+
+                root/
+                ├── class1
+                │   ├── clip1.avi
+                │   ├── clip2.avi
+                │   ├── clip3.mp4
+                │   └── ...
+                └── class2
+                    ├── clipx.avi
+                    └── ...
+
+        frames_per_clip (int): number of frames in a clip
+        step_between_clips (int): number of frames between each clip
+        transform (callable, optional): A function/transform that  takes in a TxHxWxC video
+            and returns a transformed version.
+
+    Returns:
+        tuple: A 3-tuple with the following entries:
+
+            - video (Tensor[T, H, W, C]): the `T` video frames
+            - audio(Tensor[K, L]): the audio frames, where `K` is the number of channels
+              and `L` is the number of points
+            - label (int): class of the video clip
+    """
+
+    def __init__(
+        self,
+        root,
+        frames_per_clip,
+        **kwargs
+    ):
+        warnings.warn(
+            "Kinetics400 is deprecated and will be removed in a future release."
+            "It was replaced by Kinetics(..., num_classes=\"400\").")
```

Comment:
Instead of allowing the new parameters (`split`, `num_classes`, `download`, `num_download_workers`) here and only fail when we reach the super class, we could do something like this:

```suggestion
        num_classes = None,
        split = None,
        download = None,
        num_download_workers = None,
        **kwargs
    ):
        warnings.warn(
            "Kinetics400 is deprecated and will be removed in a future release."
            "It was replaced by Kinetics(..., num_classes=\"400\").")
        if any(value is not None for value in (num_classes, split, download, num_download_workers)):
            raise RuntimeError(
                "Usage of 'num_classes', 'split', 'download', or 'num_download_workers' is not supported in Kinetics400. "
                "Please use Kinetics instead."
            )
```